scripts/dex-router/src/types.ts

```diff
@@ -0,0 +1,19 @@
+import { Connection, PublicKey, TransactionInstruction } from "@solana/web3.js";
+
+export type RouterOutput = {
+  ixs: TransactionInstruction[];
+  amountIn: bigint;
+  amountOut: bigint;
+  lookupTableAddresses?: PublicKey[];
+};
+
+export interface Router {
+  route: (
+    chainId: string,
```

Comment:
I would put `chainId`, `executor`, and `connection` inside the constructor of router and exclude them from this interface since they tend to not change during a single run.